sys/linux/dev_msr.txt

```diff
@@ -0,0 +1,15 @@
+# Copyright 2021 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+#include <asm/msr.h>
+#include <uapi/asm/msr.h>
```

Comment:
@dvyukov Added it already here. still fails for me locally. I think, maybe something is broken with my setup?

Reply:
I don't see constants added, the const file still contains no values for these constants:
https://github.com/google/syzkaller/pull/2822/files#diff-fa2f07f00f3867f6faa431d0bd2fff156ebb8c5cca3276f8e9a111dc3039d415
```
X86_IOC_RDMSR_REGS = ???
X86_IOC_WRMSR_REGS = ???
```

Reply:
I assumed the `.const` files are automatically generated (and populated) when  the corresponding header files are added to the `.txt` files when we do `make extract`. Is my assumption incorrect? do we still have to manually patch in certain constants in the auto-generated file?

Reply:
Yes, const files are auto-generated and must not be edited manually.

I've looked in detail.
First, these are not includes, these are comments. You need:
```
-#include <asm/msr.h>
-#include <uapi/asm/msr.h>
-#include <uapi/linux/fcntl.h>
+include <uapi/asm/msr.h>
+include <uapi/linux/fcntl.h>
```

Second, you will also need this because the header is not present on all arches:
```
diff --git a/sys/syz-extract/linux.go b/sys/syz-extract/linux.go
index e4f8d1bfe9b0..680111c6df20 100644
--- a/sys/syz-extract/linux.go
+++ b/sys/syz-extract/linux.go
@@ -65,6 +65,7 @@ func (*linux) prepareArch(arch *Arch) error {
                "asm/a.out.h": "",
                "asm/prctl.h": "",
                "asm/mce.h":   "",
+               "uapi/asm/msr.h": "",
        } {
```

Third, you will also need this to generate consts on the latest kernel, otherwise it fails due to some renamed consts and missing includes:
https://github.com/google/syzkaller/pull/2875

Reply:
sounds good. Thank you. Will fix and push